src/lib.rs

```diff
@@ -226,16 +224,19 @@ impl Plugin for Triforce {
 
         // Update the covariance matrix. We use an overlapping window to smooth over
         // the transitions.
-        if self.last_update.elapsed().unwrap().as_millis() > *ports.t_win as u128 {
+        if self.samples_since_last_update as f32 >= (*ports.t_win / 1000.0) * self.sample_rate {
+            self.samples_since_last_update = 0;
             self.covar_window[0].extend_from_slice(&inputs[0][0..767]);
             self.covar_window[1].extend_from_slice(&inputs[1][0..767]);
             self.covar_window[2].extend_from_slice(&inputs[2][0..767]);
             self.covar = covariance(&self.covar_window);
-            self.last_update = SystemTime::now();
             self.covar_window[0] = inputs[0][768..1023].to_vec();
             self.covar_window[1] = inputs[1][768..1023].to_vec();
             self.covar_window[2] = inputs[2][768..1023].to_vec();
         }
+        else {
+            self.samples_since_last_update += num_samples as u32;
```

Comment:
Probably worth declaring num_samples as a `u32` instead of casting it here, or make `samples_since_last_update` a `usize`.